app/views/apps/index.html.erb

```diff
@@ -1,17 +1,21 @@
-<div class="row">
-  <div class="col-md-9">
-    <p class="lead">Cluster access applications are available from the dropdown menus in the navbar above.
-    <br>Custom shared apps are available below.</p>
-    <%= render partial: "catalog_link" %>
-  </div>
-</div>
+<ol class="breadcrumb">
+  <li>
+    <%= link_to "Home", root_path %>
+  </li>
+  <li class="active">
+    All Apps
+  </li>
+</ol>
 
 <div class="row">
-  <div class="col-md-9">
-    <%= render(partial: "group", collection: @groups) || "<strong>No custom shared apps available.</strong>".html_safe %>
-  </div>
   <div class="col-md-3">
-    <%# This assumes @motd_file responds to `to_partial_path` %>
-    <%= render @motd if @motd %>
+    <div class="panel panel-default">
+      <div class="panel-heading">System Apps</div>
+      <div class="panel-body">
+        <%= render partial: "app_list", collection: @core_app_groups %>
+      </div>
+    </div>
   </div>
+
+  <%= render partial: "app_list", collection: @app_groups, layout: "col3_layout" %>
```

Comment:
Not a bug, but we may need to be aware that if `@app_groups` grows beyond 3 groups, then we will overflow the Bootstrap row. At that point we may need to explore a more dynamic grid system.

Reply:
I was betting on right now that the Bootstrap grid will just put extra columns on a separate row i.e. like it does with thumbnails:

>If more than 12 columns are placed within a single row, each group of extra columns will, as one unit, wrap onto a new line.

But the result is that produces an undesirable affect when one column is much larger that the others. The new row of columns starts after the longest column completes. Using something like https://masonry.desandro.com/ would make the columns fit better, but then they are not reliably in the same location every time.

At this point, a smaller table view, with one app per row, and a small row, might be preferable, with a text box for filtering i.e. datatables.

Benefit of having a separate Apps page is we can play with this, even provide different views  to switch between, without risking a jarring change to most user's day to day experience or affecting Open OnDemand (since changes to this doesn't affect the nav bar).

Reply:
Of course the problem with a table view is you don't see as many apps at a time.

Reply:
FWIW I think it can actually fit 4 groups. Here was a test I did earlier but didn't try this after the change I made. This is 1024x768:

![screen 2017-12-12 at 3 52 17 pm](https://user-images.githubusercontent.com/512333/34127425-19622d10-e40b-11e7-8184-207a573a542d.png)

Reply:
Masonry is actually nice since we have some very long columns and short ones. Also, how does it handle 5 columns, if there are 4 `@app_groups` + the `@core_app_groups`?